### PR TITLE
Add support for AD7291

### PR DESCRIPTION
**Pull request by mthoren-adi**
# Description

Add support for AD7291 8-channel, I2C, 12-bit SAR ADC with temperature sensor used on multiple FMC cards and CftLs

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Tested on CN0566 / Phaser setup. Output from example script:

Temperature:  36.5  C
Channel 0:  914.916991438  millivolts
Channel 1:  1513.67187376  millivolts
Channel 2:  1641.84570178  millivolts
Channel 3:  1131.591795948  millivolts
Channel 4:  1.220703124  millivolts
Channel 5:  1271.9726552079999  millivolts
Channel 6:  927.73437424  millivolts
Channel 7:  567.0166010979999  millivolts

**Test Configuration**:
* Hardware: CN0566 Phaser board, Raspberry Pi 4
* OS: ADI Kuiper Linux on remote Pi, Pyadi-iio tested on a remote Windows 10 machine

# Documentation

retstring:
ad7291(uri="ip:phaser.local") object ""
8-channel, I2C, 12-bit SAR ADC with temperature sensor

Channel layout:

voltageX.raw:              Raw 12-bit ADC code. read only for ADC channels
voltageX.scale:            ADC scale, millivolts per lsb
voltageX():                    Returns ADC reading in millivolts (read only)

temp0.raw:                      Temperature raw value
temp0.scale:                    Temperature scale value
temp0():                        Returns temperature in degrees Celsius

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mthoren-adi** [comment]
@PopPaul2021 tagging you because you're using this device in lldk_loopback_example_rebase. The channel layout was a bit annoying,  because channel[0] corresponds to temperature, channel[1] to input 0, etc. This is a flat layout, similar to AD5592r.
@tfcollins one question - this device has no writeable attributes; does it need a pytest fixture? I see a attribute_write_only_str function, but no read_only.

File: adi/ad7291.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        @property
+        def raw(self):
+            """AD7291 channel raw value"""
+            return self._get_iio_attr(self.name, "raw", False)
+
+        @property
+        def mean_raw(self):
+            """AD7291 channel mean_raw value"""
+            return self._get_iio_attr(self.name, "raw", False)
```
Comment (tfcollins) [inline adi/ad7291.py:124]: These the same?

**Review by tfcollins — COMMENTED** [review]
Can you add a test?

File: adi/ad7291.py (hunk 1 above)
Comment (mthoren-adi) [reply adi/ad7291.py]: Yup, that was a bug. I also made _temp_channel a subclass... probably not necessary but a tiny bit neater.

**Comment by mthoren-adi** [comment]
> Can you add a test?

Ideally this would be implemented as a attribute_single_value_readonly() function in attr_tests, making use of dev_interface_sub_channel() with an added option for read-only. But it looks like there are some issues with dev_interface_sub_channel() (and dev_interface()) that should be addressed separately.

**Pull request merged**